### ISSUE TITLE
Source Control > Discard All Files not working properly

### ISSUE THREAD

**Issue by pellea** [Microsoft/vscode#44272]
- VSCode Version: 1.2.1.0-insider
- OS Version: Win 10

Steps to Reproduce:

1. Create one unversioned file and edit 2 other files (for example)
2. Go to "Source Control"
3. Go to "..." => "Discard all changes"
4. On the popup choose the "discard all 3 files" button

Only the unversioned file is removed. I have to do it again to discard the 2 other versioned changes.

**Comment by joaomoreno** [comment]
Is there anything special about your setup? Files have weird names? Is it a network share? Can you reproduce on other workspaces?

![kapture 2018-02-26 at 11 00 24](https://user-images.githubusercontent.com/22350/36664223-48b0e66e-1ae4-11e8-8af1-787d3bd0294d.gif)

**Comment by pellea** [comment]
Here are some information about the workspace:

- The repository uses Git with VS Team Services. 
- I have the .gitignore from https://github.com/github/gitignore/blob/master/VisualStudio.gitignore
- The app I'm working on is a React app created with create-react-app
- I have the VS Team extension for VSCode installed and configured for the repo (https://marketplace.visualstudio.com/items?itemName=ms-vsts.team)

I don't have any others workspaces that uses vscode (the others use Visual Studio). I can create a new dummy project on VSTS and try there if you want.

**Comment by joaomoreno** [comment]
@pellea Yes please do.

**Comment by pellea** [comment]
Here are the complete steps to the bug:

1. New Project in VSTS (Scrum template)
2. Initialize wtih a README or gitignore (only gitignore)
3. PS> git clone https://xxxx.visualstudio.com/_git/VSCodeDebug vscodedebug
4. PS> cd vscodedebug
5. PS> create-react-app . --scripts-version=react-scripts-ts
6. PS> explorer .      && right click open vscode here
7. Source control > "initial commit" > ctrl+enter
8. Source control > ... > push to ... > select remote repo > enter
9. See the beautiful Gif video for the remaining part :)

![vscodedebug](https://user-images.githubusercontent.com/395982/36686630-54607ab8-1b27-11e8-8fdb-91520a74e1a0.gif)

**Comment by joaomoreno** [comment]
After the first discard, can you click the `App.tsx` change to see what changes appear?

**Comment by pellea** [comment]
There is no changes. The file has been reverted.

**Comment by joaomoreno** [comment]
But the SCM viewlet still shows that the file has a change. If you click it, Code doesn't show any change? What does `git diff -- App.tsx` return?

**Comment by pellea** [comment]
Yes the SCM viewlet shows a change.

`git diff` returns nothing

`git status` shows 
```

PS D:\dev\vscodedebug> git status
On branch master
Your branch is up to date with 'origin/master'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

        modified:   src/App.tsx

no changes added to commit (use "git add" and/or "git commit -a")
PS D:\dev\vscodedebug>
```

and after discarding the second time:

```
PS D:\dev\vscodedebug> git status
On branch master
Your branch is up to date with 'origin/master'.

nothing to commit, working tree clean
PS D:\dev\vscodedebug>
```

**Comment by joaomoreno** [comment]
Hm... `git status` says the file is dirty, but `git diff` returns no changes. Sounds like some [line ending issues](https://help.github.com/articles/dealing-with-line-endings/). What are your `core.autocrlf` and `core.safecrlf` settings?

**Comment by pellea** [comment]
git config core.autocrlf => true
git config core.safecrlf => nothing

**Comment by joaomoreno** [comment]
Can you change `core.autocrlf` to false and try again?

**Comment by pellea** [comment]
That doesn't seem like a good idea without breaking every projects I own on the machine...

> 
> 
> After you've set the core.autocrlf option and committed a .gitattributes file, you may find that Git wants to commit files that you have not modified. At this point, Git is eager to change the line endings of every file for you.
> 
> The best way to automatically configure your repository's line endings is to first backup your files with Git, delete every file in your repository (except the .git directory), and then restore the files all at once.

**Comment by joaomoreno** [comment]
I'm just interested in knowing whether that is what actually causes the issue. You can switch it back once you confirm it.

**Comment by pellea** [comment]
I executed the command "git config --global core.autocrlf false" and it does the same thing. I have to discard twice. (git status show the file as modified)

**Comment by joaomoreno** [comment]
Does it happen too if you run with `code --disable-extensions`?

**Comment by pellea** [comment]
@joaomoreno same behavior

**Comment by joaomoreno** [comment]
Can you show me what comes up in `Git: Show Git Output` when you run the discard action?

**Comment by pellea** [comment]
Here is the log:

```
> git clean -f -q -- d:\dev\vscodedebug\src\Test.tsx
> git checkout -q -- d:\dev\vscodedebug\src\App.tsx
> git status -z -u
> git symbolic-ref --short HEAD
> git rev-parse master
> git show :src/App.tsx
> git rev-parse --symbolic-full-name --abbrev-ref master@{u}
> git rev-list --left-right master...origin/master
> git for-each-ref --format %(refname) %(objectname) --sort -committerdate
> git remote --verbose
```

**Comment by joaomoreno** [comment]
OK, can you try running those same commands in the command line?

```
git clean -f -q -- d:\dev\vscodedebug\src\Test.tsx
git checkout -q -- d:\dev\vscodedebug\src\App.tsx
```

What is the outcome?

**Comment by joaomoreno** [comment]
So what happens if you run the command as it is, with the absolute path?

**Comment by pellea** [comment]
Sorry ... the second command is nor working. If I want to remove the App.tsx file I need to run it twice.

It seems the issue I have is actually not related to VSCode.

**Comment by joaomoreno** [comment]
Got it! Let us know if you ever find a fix, I'd be interested in knowing what it is.

**Comment by pellea** [comment]
I updated the git tools to version 2.16.2.
The issue is fixed (I had the 2.16.0 version).
